### PR TITLE
Document Option class

### PR DESCRIPTION
**Pull request by honzatrtik**
+ add handy `Option::fromNullable` constructor

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,6 +162,84 @@ $sumThree(7, 42, 666);
 
 All invocations will yield the same result.
 
+#### Avoid billion dollar mistake by using `Option`
```
Comment (jost125) [inline README.md]: Not every body knows, that `null` reference is called "billion dollar mistake". Maybe change this to:

```
Avoid billion dollar mistake (`null`) by using `Option`
```

File: README.md [hunk 2]
```diff
@@ -162,6 +162,84 @@ $sumThree(7, 42, 666);
...
+   
+    }
+    if ($age === null) {
+        print "Dont know age of user with id {$id}";
+    } else {
+        print "Age of user with id {$id} is {$age}";
+    }   
+     
+}
+
```
Comment (jost125) [inline README.md:230]: I would split this code snippet into two and first one I would label with "classic approach with null" and second one "compare with functional approach with Option". Or something like that

File: README.md (hunk 1 above)
Comment (jost125) [reply README.md]: Please use `###` level of headline and add it into TOC.

File: README.md (hunk 2 above)
Comment (honzatrtik) [reply README.md:230]: I used to have it that way :-D 

Will revert it...

**Pull request merged**